py/htm/encoders/date.py

```diff
@@ -299,7 +300,7 @@ def encode(self, inp, output=None):
             break
           elif diff.days == 1:
             # ramp smoothly from 1 -> 0 on the next day
-            val = 1.0 - (float(diff.seconds) / 86400)
+            val = 1.0 + (float(diff.seconds) / 86400)
```

Comment:
Are you sure about this change?

Reply:
Well, I tried to explain this in issue #762 .

It could be argued that the slope up and the slope down on either side of the holiday should be the same semantics and therefore should be the same.  However, my thinking is that the semantics should follow the same logic as any other encoding with a wrap.  The pattern for the down slope should shift to the right (not the left) for advancing time.
```
    notholiday = [1,1,1,1,1,0,0,0,0,0]  # Not a holiday
    holiday1   = [0,0,0,1,1,1,1,1,0,0]  # day before holiday
    holiday    = [0,0,0,0,0,1,1,1,1,1]  # day of holiday
    holiday2   = [1,1,0,0,0,0,0,1,1,1]  # day after holiday; This was  [0,0,0,1,1,1,1,1,0,0]
```

Reply:
OK, that makes a lot of sense.  With your change: the time before & after a holiday are semantically different.  Thanks for the explanation.